NTupleMaker/bin/SynchNTupleProducer_2017.cpp

```diff
@@ -797,7 +779,9 @@ for (Long64_t iEntry=0; iEntry<numberOfEntries; iEntry++) {
       if(ch == "mt"){
         for (unsigned int im = 0; im<analysisTree.muon_count; ++im) {
 
-	  bool muonMediumId = isIdentifiedMediumMuon(im, &analysisTree, isData);
+	  bool muonMediumId=true;
+	  if(applyMuonId){//2019 8 8: Merijn: impose that the flag from the config file is used
+	    muonMediumId= isIdentifiedMediumMuon(im, &analysisTree, isData);}
   			
```

Comment:
Is there any circumstance when we do not want to check the muon identification?

Reply:
Hi Andrea,
I cannot answer from an analysis perspective, i.e. if there are physics analyses that don't want to check on muon identification.

However, we define a flag in the config file to y/n check upon applyMuonId.
In the old version of the code, this flag is not used anywhere. This is confusing in my opinion; a flag should either be actually used, or not defined at all.

Thus I implemented what seems most logic to me: per default we take the muon. If the flag indicates that we need to check, then the boolean can be changed. In case the flag doesn't impose to cross check, treat as if the muon passed.

Reply:
I think you always want to apply the muon id. Thus, from my point of view the bool is not needed and can be removed. But you can of course also keep it.